### PR TITLE
Creating configs for soon to be main branch of odo

### PR DESCRIPTION
**Pull request by mohammedzee1000**
- this does not remove existing configs for master.
   that will happen is seperate prs once switch is
   complete and verified

Signed-off-by: Mohammed Zeeshan Ahmed <mohammed.zee1000@gmail.com>

### CODE REVIEW

File: ci-operator/config/openshift/odo/openshift-odo-main__v4.7.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    cluster_profile: aws
+    test:
+    - as: integration-e2e-periodic-steps
+      commands: scripts/openshiftci-periodic-tests.sh
+      from: oc-bin-image
+      resources:
+        requests:
+          cpu: "2"
+          memory: 6Gi
+      timeout: 4h0m0s
```
Comment (prietyc123) [inline ci-operator/config/openshift/odo/openshift-odo-main__v4.7.yaml:60]: Why are we not adding timeouts `4h0m0s` for 4.5 periodic job

File: ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    skip_cloning: true
+  extra_refs:
+  - base_ref: main
+    org: openshift
+    repo: odo
+  labels:
+    ci-operator.openshift.io/prowgen-controlled: "true"
+    ci-operator.openshift.io/variant: v4.5
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-odo-main-v4.5-integration-e2e-periodic
```
Comment (prietyc123) [inline ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml:16]: @mohammedzee1000 I think you missed adding `reporter_config` for periodic jobs https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/odo/openshift-odo-master-periodics.yaml#L17-L27
Could you please add it for all the periodic jobs?

File: ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml [hunk 3]
```diff
@@ -0,0 +1,369 @@
+presubmits:
+  openshift/odo:
+  - agent: kubernetes
+    always_run: true
```
Comment (prietyc123) [inline ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml:4]: I think always run should be `false` in our case. Correct me if I am wrong.

File: ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci-operator.openshift.io/prowgen-controlled: "true"
+      ci-operator.openshift.io/variant: psi
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-odo-main-psi-kubernetes-integration-e2e
+    optional: true
+    rerun_command: /test psi-kubernetes-integration-e2e
```
Comment (prietyc123) [inline ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml:18]: We need to add `run_if_changed` field for all the prow jobs running on pr, no? https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/odo/openshift-odo-master-presubmits.yaml#L19

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml (hunk 2 above)
Comment (mohammedzee1000) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml:16]: Added, will push shortly

File: ci-operator/config/openshift/odo/openshift-odo-main__v4.7.yaml (hunk 1 above)
Comment (mohammedzee1000) [reply ci-operator/config/openshift/odo/openshift-odo-main__v4.7.yaml:60]: Added, pushing shortly

File: ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml (hunk 2 above)
Comment (prietyc123) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml:16]: Still its not reflected 🙁 . Actually when we do `make jobs` it removes the extra config we need. So it needs to be added once we are done with `make jobs`

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml (hunk 2 above)
Comment (prietyc123) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml:16]: Ahh my bad  I thought you already pushed the changes.

**Comment by mohammedzee1000** [comment]
Ok i see that only repoter configs got removed with make jobs luckily commited all changes before running make jobs so il checkout and push

File: ci-operator/config/openshift/odo/openshift-odo-main__v4.7.yaml (hunk 1 above)
Comment (mohammedzee1000) [reply ci-operator/config/openshift/odo/openshift-odo-main__v4.7.yaml:60]: added

File: ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml (hunk 3 above)
Comment (mohammedzee1000) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml]: done

File: ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml (hunk 4 above)
Comment (mohammedzee1000) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml:18]: done

File: ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+  reporter_config:arout, 10 months ago: • Send periodic job notification to CoreOS slac…
+    slack:
+      channel: '#odo-notifications'
+      job_states_to_report:arout, 9 months ago: • Push notifications types and template
```
Comment (prietyc123) [inline ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml:20]: Why the fields are looking like a statement? Am I missing something here 😕

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml (hunk 5 above)
Comment (mohammedzee1000) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml:20]: hmm

File: ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci-operator.openshift.io/prowgen-controlled: "true"
+      ci-operator.openshift.io/variant: v4.5
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-odo-main-v4.5-images
+    rerun_command: /test v4.5-images
+    run_if_changed: ^(?:[^d]+|d(?:$|[^o]|o(?:$|[^c]|c(?:$|[^s]))))*$
```
Comment (prietyc123) [inline ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml:119]: I think we don't need `run_if_changed` to be added in`ci/prow/v4.5-images` as we are not at all running these images on pr. `run_if_changed` should only be applied to `unit`, `ci/prow/psi-kubernetes-integration-e2e` and `ci/prow/v4.7-integration-e2e`, no?

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml (hunk 5 above)
Comment (mohammedzee1000) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-periodics.yaml]: fixed

File: ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci-operator.openshift.io/prowgen-controlled: "true"
+      ci-operator.openshift.io/variant: v4.8
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-odo-main-v4.8-images
+    rerun_command: /test v4.8-images
+    run_if_changed: ^(?:[^d]+|d(?:$|[^o]|o(?:$|[^c]|c(?:$|[^s]))))*$
```
Comment (prietyc123) [inline ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml:338]: I think you missed removing it 🙂

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml (hunk 7 above)
Comment (mohammedzee1000) [reply ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml]: removed

**Pull request merged**